### PR TITLE
Added DEM support to heightmaps

### PR DESCRIPTION
**Pull request by jennuine**
# 🎉 New feature

* Part of https://github.com/ignitionrobotics/ign-gazebo/issues/235
* Requires: 
  * ~~https://github.com/ignitionrobotics/ign-common/pull/267~~
  * https://github.com/ignition-tooling/release-tools/issues/567
  * https://github.com/ignitionrobotics/ign-common/pull/292

## Summary
Extended heightmap implementation to support DEMs.

`ign-gazebo` preview:
Peek 2021-12-15 19-14

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by chapulina** [comment]
* Yay, nice! Also requires https://github.com/ignition-tooling/release-tools/issues/567 so this branch is used by Garden.

File: dartsim/src/CustomHeightmapShape.cc [hunk 1]
```diff
@@ -33,37 +34,52 @@ CustomHeightmapShape::CustomHeightmapShape(
...
     scale.Z(1.0);
   else
     scale.Z(fabs(_size(2)) / heightmapSizeZ);
 
   auto sizeIgn = ignition::math::eigen3::convert(_size);
 
-  // We need to make a copy here in order to use the non-const FillHeightMap
+  std::vector<float> heightsFloat;
+
+  // We need to make a copy below in order to use the non-const FillHeightMap
```
Comment (scpeters) [inline dartsim/src/CustomHeightmapShape.cc:54]: I just noticed this complaint, so I made the `FillHeightmap` function `const` in https://github.com/ignitionrobotics/ign-common/pull/292/commits/8c63dac5e9ecc7495e33ca65657b6ed12d7ece18 in https://github.com/ignitionrobotics/ign-common/pull/292

I think we can skip copying the data with this change

Comment (jennuine) [reply dartsim/src/CustomHeightmapShape.cc]: Thanks! d78cd486d264ebe122e58566a7ef243316f5dba6

**Comment by scpeters** [comment]
CI is giving warnings because it is not using https://github.com/ignitionrobotics/ign-common/pull/292

**Review by chapulina — APPROVED** [review]
Nice, LGTM!

**Pull request merged**